code/model/sudoku/SudokuPuzzle.java

```diff
@@ -0,0 +1,116 @@
+package code.model.sudoku;
+
+import java.util.Stack;
+import code.model.*;
+
+/**
+ * Sudoku puzzle class
+ */
+public class SudokuPuzzle {
+
+    private int[][] numbers;
+    private boolean[][] numberLocks;
+    private int[][] defaultAnswer;
+    private int selectedNumber;
+    private Stack<SudokuMove> moveHistory;
+
+    /**
+     * Default Constructor
+     */
+    public SudokuPuzzle() {
+        this.numbers = new int[9][9];
+        this.numberLocks = new boolean[9][9];
+        this.defaultAnswer = new int[9][9];
+        this.selectedNumber = null;
```

Comment:
Unless I'm missing something, you shouldn't be able to set an `int` to `null` in Java.
Recommend using a default value, such as 0 or -1.